src/target/riscv_debug.c

```diff
@@ -999,7 +1047,34 @@ static size_t riscv_build_target_description(
 	offset += snprintf(buffer + offset, print_size, "<reg name=\"pc\" bitsize=\"%u\"%s/>", address_width,
 		gdb_reg_type_strings[GDB_TYPE_CODE_PTR]);
 
-	/* XXX: TODO - implement generation of the FPU feature and registers */
+	
+
+	/* Basic single precision support */
+	if (extensions & RV_ISA_EXT_SINGLE_FLOAT) {
+		const int first_fpu_register = RV_FPU_GDB_FIRST_REGISTER;         // fixme!
+		const int first_fpu_control_register = RV_FPU_GDB_CSR_REGISTER; // fixme!
+
+		if (max_length != 0)
+			print_size = max_length - (size_t)offset;
+		offset += snprintf(buffer + offset, print_size, "</feature><feature name=\"org.gnu.gdb.riscv.fpu\">");
+
+		for (unsigned int i = 0; i < sizeof(riscv_fpu_regs) / sizeof(riscv_fpu_regs[0]); i++) {
+			if (max_length != 0)
+				print_size = max_length - (size_t)offset;
+			offset += snprintf(buffer + offset, print_size,
+				"<reg name=\"f%s\" bitsize=\"32\" type=\"ieee_single\" regnum=\"%u\" group=\"float\"/>",
```

Comment:
One notable thing from our prior work with GDB here, we can likely drop both the `group=` and `type=` stanzas which reduces the size of the feature blob generated here, which can otherwise get rather large. This reduces the likelihood of an OOM condition occurring on STM32F1-based probes.